EIPS/eip-1011.md

```diff
@@ -0,0 +1,287 @@
+```python
+casper.last_justified_epoch() == casper.current_epoch()
+```
+
+Clients should only consider checkpoints justified or finalized if deposits were greater than `NON_REVERT_MIN_DEPOSIT` _during the epoch in question_\*:
+```python
+casper_state_during_epoch_in_question.total_curdyn_deposits_scaled() > NON_REVERT_MIN_DEPOSIT
+casper_state_during_epoch_in_question.total_prevdyn_deposits_scaled() > NON_REVERT_MIN_DEPOSIT
+```
+
+\* When assessing the size of total deposits for epoch finalization and justification, use `total_*dyn_deposits_scaled` from the state of the VM during the epoch. For example, if assessing whether epoch 10 should be finalized and epoch 10 occurred from block 500 to 550, then get the `total_*dyn_deposits_scaled` during that 500 to 550 block range rather than during the current epoch.
+
+#### Block Reward
+
+If `block.number >= HYBRID_CASPER_FORK_BLKNUM`, then `block_reward = NEW_BLOCK_REWARD` and utilize the same formulas for ommer rewards but with the updated `block_reward`.
+
+#### Validators
+
+The mechanics and responsibilities of validators are not specified in this EIP because they rely upon network transactions to the contract at `CASPER_ADDR` rather than on protocol level implementation and changes.
+See the [Validator Implementation Guide](https://github.com/ethereum/casper/blob/master/VALIDATOR_GUIDE.md) for validator details.
+
+#### SIGHASHER_CODE
+
+The source code for `SIGHASHER_CODE` is located [here](https://github.com/ethereum/casper/blob/master/casper/contracts/sighash.se.py).
+The source is to be migrated to Vyper LLL before it's bytecode is finalized for this EIP.
+
+The EVM init code is:
+```
+TBD
+```
+
+The EVM bytecode that the contract should be set to is:
+```
+TBD
+```
+
+#### PURITY_CHECKER_CODE
+
+The source code for `PURITY_CHECKER_CODE` is located [here](https://github.com/ethereum/research/blob/master/impurity/check_for_impurity.se).
+The source is to be migrated to Vyper LLL before it's bytecode is finalized for this EIP.
+
+The EVM init code is:
+```
+TBD
+```
+
+The EVM bytecode that the contract should be set to is:
+```
+TBD
+```
+
+#### CASPER_CODE
+
+The source code for `CASPER_CODE` is located at
+[here](https://github.com/ethereum/casper/blob/master/casper/contracts/simple_casper.v.py).
+The contract is to be formally verified and further tested before it's bytecode is finalized for this EIP.
+
+The EVM init code with the above specified params is:
+```
+TBD
+```
+
+The EVM bytecode that the contract should be set to is:
+```
+TBD
+```
+
+#### Client Settings
+Clients should be implemented with the following configurable settings:
+
+##### Enable Casper Fork Choice
+The ability to enable/disable the Casper Fork Choice. A suggested implementation is `--casper-fork-choice`.
+
+This setting should ship as default disabled in client versions during the initial casper fork. This setting should ship as default enabled in subsequent client versions.
```

Comment:
What's the motivation behind this? We'll have the nodes with different fork choice rules active on the net, which seems to increase the probability of the unintentional fork?
Why give the user an option to change consensus rules at all?

Reply:
The two fork choice rules can only diverge if either 51% of miners or 51% of validators are faulty.

The reason why we need to give clients a choice is to make it easier to revert back to PoW fork choice in case there is any issue with the PoS consensus in the short term.